### PR TITLE
kern_mmap cleanup

### PR DESCRIPTION
**Pull request by brooksdavis**
A couple cleanups inspired by a chat with @jrtc27 a while ago. I plan to upstream the first two patches. I'm slightly conflicted about splitting them. What I've done it easier to describe this way, but will result in more conflicts when looping back later.

### CODE REVIEW

File: sys/compat/linux/linux_mmap.c [hunk 1]
```diff
@@ -217,7 +217,7 @@ linux_mmap_common(struct thread *td, uintptr_t addr, size_t len, int prot,
 	    (bsd_flags & MAP_EXCL) == 0) {
 		mr_fixed = mr;
 		mr_fixed.mr_flags |= MAP_FIXED | MAP_EXCL;
-		error = kern_mmap_req(td, &mr_fixed);
+		error = kern_mmap(td, &mr_fixed);
 		if (error == 0)
 			goto out;
 	}
```
Comment (jrtc27) [inline sys/compat/linux/linux_mmap.c:223]: One just below you've missed

Comment (brooksdavis) [reply sys/compat/linux/linux_mmap.c:223]: bah, fixed three

**Review by bsdjhb — APPROVED** [review]
Merging kern_mmap_req/kern_mmap is a good cleanup.

The compound literal is a bit new relative to the rest of the kernel's code-base (I don't know of any other uses in the tree of this fashion).

**Comment by jrtc27** [comment]
> The compound literal is a bit new relative to the rest of the kernel's code-base (I don't know of any other uses in the tree of this fashion).
```
jrtc4@zeno:~/cheri/cheribsd$ git grep '&(struct' upstream/main sys/ | wc -l
      36
```

**Comment by arichardson** [comment]
> > The compound literal is a bit new relative to the rest of the kernel's code-base (I don't know of any other uses in the tree of this fashion).
> 
> ```
> jrtc4@zeno:~/cheri/cheribsd$ git grep '&(struct' upstream/main sys/ | wc -l
>       36
> ```

There are a few other uses in userland and all supported compiler can handle it. Seems fine to me.

**Comment by bsdjhb** [comment]
Hmm, those cases are all very new (and in niche parts of the code base hardly anyone else looks at).  sys/stats.h has some truly horrible macros (not the ones using the compound literal, but the ones that define individual sub members you would use inside of a literal).  It is a bit conciser than what you would otherwise do which is to just init a local var with the same initializer block.  While it is reminiscent of a C++ lambda in syntax to me, in actual C++ you'd just use a constructor that took the arguments instead.

**Comment by brooksdavis** [comment]
Compound literals came in with C99 (along with designated initializers) so no compiler support issues. I think they fit this use case well and avoid the need for arguments about where to declare variables that would arise if we wanted to use designated initializers with non-static values. I'm not 100% sure of my style choices.

**Comment by bsdjhb** [comment]
To be clear, I approved the commit. :)

**Pull request merged**